docs/ibc/middleware/develop.md

```diff
@@ -239,4 +240,21 @@ func SendPacket(appPacket channeltypes.Packet) {
 
     return ics4Keeper.SendPacket(packet)
 }
+
+// middleware must unwrap the channel version so the underlying application 
+// which calls this function receives the version it constructed.
+func GetUnwrappedChannelVersion(ctx sdk.Context, portID, channelID string) (string, bool) {
+    version, found := ics4Keeper.GetUnwrappedChannelVersion(ctx, portID, channelID)
+    if !found {
+        return "", false
+    }
+
+    // unwrap channel version
+    metadata, err := Unmarshal(version)
+    if err != nil {
+        panic() 
```

Comment:
Should probably include the err in the panic here

Reply:
Should not panic at all and instead return error.